### ISSUE TITLE
🏁 Issue v23.6.0

### ISSUE THREAD

**Issue by andybak** [itchio/itch#1577]
Dear itch app team, 

After reinstalling Windows I was hoping that pointing the itch app to my previous itch games directory would restore my library (much as Steam does).

**Comment by fasterthanlime** [comment]
That's not the case right now, but it's planned for future versions of the app to behave like that.

**Comment by fasterthanlime** [comment]
v25 is out, and it does that.

The only thing it won't do is:

  * if your library was installed with a previous version of the app (v23 or earlier)
  * %APPDATA% was wiped (after a complete reinstall for example)
  * it's being added to a v25 install again

However, if the library is installed with v25, then a complete reinstall occurs, then it's added again, then it *will* work.

**Comment by andybak** [comment]
> %APPDATA% was wiped (after a complete reinstall for example)

Do you mean literally %APPDATA% or "where ever I've decided to store my Itch library"? Because a "complete reinstall" is when I most want my library to be restored. I have an automated backup that covers my Itch library - but it's not in %APPDATA%.

To be honest - I'm confused why you're not restoring directly from my online account details. I presume you know what I've downloaded. Why not provide the ability to restore this list?

**Comment by fasterthanlime** [comment]
> Do you mean literally %APPDATA%

For itch v23, `%APPDATA%\itch\marketdb`.

I fully understand the frustration - not all scenarios were covered before v25.

The app needs both:

  * A local database that contains all games, uploads, builds, users, installed game info, etc. that's fast to query and sort, filter, search (for v23 that was `marketdb`, for v25 it's an SQLite database in `db`)
  * A receipt in each installed game folder, in case the libraries are moved across computers and need to be re-imported (v23 did write receipts, but didn't normalize games installed with even older versions).

> I'm confused why you're not restoring directly from my online account details

I'm not entirely sure we keep track of everything that's downloaded and by who. It definitely increments analytics, but they're not, for example, automatically added to a collection (see #2023). Even if it was, we wouldn't be able to tell apart which computers you installed which games on.

---

tl;dr - I'm sorry for the mess, it took a while to fully understand how to solve all scenarios. v25 is strict about what it imports precisely to make sure that it is (from this point), future-proof. It has a best-effort import routines for versions of the app as old as v18 (see [the code](https://github.com/itchio/butler/blob/master/endpoints/install/locations_scan.go)), but it always fetches fresh data from the API, so we don't end up with malformed data in the database (which would be much worse).
